### PR TITLE
Allow alias dereferencing in LDAP searches

### PR DESCRIPTION
**Pull request by f4z3r**
Implements the requested improvement from #6128

### CODE REVIEW

File: website/content/docs/auth/ldap.mdx [hunk 1]
```diff
@@ -133,6 +133,10 @@ There are two alternate methods of resolving the user object used to authenticat
 
 @include 'ldap-auth-userfilter-warning.mdx'
 
+#### Alias Dereferencing
+
+- `dereference_aliases` (string, optional) - When to dereference aliases when performing the search. Possible values are: `never`, `finding`, `searching`, and `always`. `finding` will only dereference aliases during name resolution of the base. `searching` will dereference aliases after name resolution.
```
Comment (tjperry07) [inline website/content/docs/auth/ldap.mdx:138]:
```suggestion
- `dereference_aliases` (string, optional) - Control how aliases are dereferenced when performing the search. Possible values are: `never`, `finding`, `searching`, and `always`. `finding` will only dereference aliases during name resolution of the base. `searching` will dereference aliases after name resolution.
```

Comment (tjperry07) [reply website/content/docs/auth/ldap.mdx:138]: Just a small wording change. Otherwise, it's great.

**Review by tjperry07 — APPROVED** [review]

**Review by tjperry07 — APPROVED** [review]

**Comment by jboero** [comment]
This does look like a good feature addition to LDAP auth.  @hashicorp/vault-admin please have a look?  Thanks

File: website/content/docs/auth/ldap.mdx [hunk 2]
```diff
@@ -133,6 +133,10 @@ There are two alternate methods of resolving the user object used to authenticat
 
 @include 'ldap-auth-userfilter-warning.mdx'
 
+#### Alias Dereferencing
+
+- `dereference_aliases` (string, optional) - Control how aliases are dereferenced when performing the search. Possible values are: `never`, `finding`, `searching`, and `always`. `finding` will only dereference aliases during name resolution of the base. `searching` will dereference aliases after name resolution.
+
```
Comment (maxcoulombe) [inline website/content/docs/auth/ldap.mdx:139]: Would it be possible to also add the new parameter to the [api-docs](https://github.com/f4z3r/vault/blob/feat/6128/website/content/api-docs/auth/ldap.mdx?plain=1#L93)?

Something like:
```
- `dereference_aliases` `(string: never)` - When aliases should be dereferenced on search operations. Accepted values are 'never', 'finding', 'searching', 'always'. Defaults to 'never'.
```

**Review by maxcoulombe — COMMENTED** [review]
Hey @f4z3r thanks for taking care of this. 

I verified if the default value is indeed "never" for backward compatibility, checked that the new parameter is passed correctly both as input and output on the API and that it is used to populate the ldap search requests where ever a search was performed and everything looks covered.

I just have a small suggestion to document the `dereference_aliases` param on the api-docs page for completeness and this would be good to ship!

File: website/content/docs/auth/ldap.mdx (hunk 2 above)
Comment (f4z3r) [reply website/content/docs/auth/ldap.mdx:139]: Sure thing. Added your proposed text in the API docs.

**Comment by maxcoulombe** [comment]
Thanks for updating the api-docs, everything looks good to me!

**Review by maxcoulombe — APPROVED** [review]

**Pull request merged**